### PR TITLE
Add Azure DevOps PR test matrix documentation

### PR DESCRIPTION
**Pull request by pedropaulovc**
## Summary

- Adds comprehensive test dataset documentation for Azure DevOps integration
- 63+ test cases across 13 PRs in the `pedropaulovc/BasicAppServiceRecommender` Azure DevOps repo
- Covers all CodjiFlo comment and iteration features

### Test Categories

| PR | Category | Test Cases |
|----|----------|------------|
| #2 | Comment Positioning | CP-01 to CP-06 |
| #3 | Threading & States | CT-01 to CT-08 |
| #4 | File Operations | FO-01 to FO-05 |
| #5 | Iteration Tracking | IT-01 to IT-10 |
| #6 | Code Suggestions | CS-01 to CS-04 |
| #7 | Top-Level Comments | TL-01 to TL-06 |
| #8 | PR State Transitions | PS-01 to PS-03 |
| #9 | Draft PR | PS-05 |
| #10 | Edge Cases | EC-01 to EC-05 |
| #11 | Cross-Line-Type | CLT-01 to CLT-03 |
| #13 | Comment Interactions | CI-01 to CI-03 |
| #14 | File Add+Rename | FAR-01 |
| #15 | Multi-Commit Push | MC-01 to MC-03 |

## Test plan

- [x] All test PRs created in Azure DevOps repo
- [x] Comments added via REST API
- [x] Documentation complete with API references

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: spec/test/azure-devops-test-matrix.md [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+### Suggestion Markdown Format
+
+```markdown
+I suggest changing this to:
+```suggestion
+const result = calculateValue(input);
+return result;
+```
+This would improve readability.
+```
```
[the remark this answers is not in the document]
Comment (pedropaulovc) [reply spec/test/azure-devops-test-matrix.md]: Fixed. Changed the nested code block example to use 4-space indentation instead of triple backticks to avoid rendering issues.

-- 🤖 [Claude Code](https://claude.ai/claude-code)
